bounded/src/Data/Bounded/TypeLevel.hs

```diff
@@ -26,23 +26,21 @@ type IsInRange (nat :: Nat) (min :: Nat) (max :: Nat)
 -- | Constructor for appending a type level string with a normal Type
 data (a :: Symbol) ++: (b :: Type)
 
-type family (++) (a :: Symbol) (b :: k) :: Symbol where
-  a ++ (b :: Symbol) = a `AppendSymbol` b
-
 typeName :: forall (a :: Type). Typeable a => String
 typeName
   = fromMaybe (error "GHC error invalid type name")
   . List.stripPrefix "Proxy * "
   . show
   $ typeOf (Proxy @a)
 
+type Length :: Symbol -> Nat
 type family Length (a :: Symbol) :: Nat where
-  Length symbol = Length' (ToList symbol)
-
-type family Length' (a :: [Symbol]) :: Nat where
-  Length' '[] = 0
-  Length' (x ': xs) = 1 + Length' xs
+  Length symbol = Length' (UnconsSymbol symbol)
 
+type Length' :: Maybe (Char, Symbol) -> Nat
```

Comment:
@epicallan Since the char kind proposal we can use `UnconsSymbol` which is much faster than what `symbols` package did.

https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0387-char-kind.rst

Eligible since 9.2. Should compile faster as it removes an intermediary.

Reply:
I like this change & thanks for the link to the proposal.